application-numbered-content-headings/application-numbered-content-headings-ui/src/main/resources/NumberedHeadings/Code/NumberedHeadings.xml

```diff
@@ -695,7 +697,90 @@ counter-reset: ch5 var(--numbered-headings-start) ch6;
 
 #buildSelector($rootSelectors 'h6[data-numbered-headings-start]') {
 counter-reset: ch6 var(--numbered-headings-start);
-}</code>
+}
+
+/* Newer browser support based on counter-set. */
+@supports (counter-set: nh1) {
```

Comment:
Difficult to validate without extensive manually testing, but lgtm.
I don't see anything related to number skipping, is it because it is simply not needed here?

Reply:
Regarding number skipping, isn't this covered by :not([data-xwiki-rendering-protected]) in most of the selectors below?

The code should be mostly identical to the code above except that:

* The initialization is on the root selectors only.
* It uses `counter-set` instead of `counter-reset`.
* Counters are increment on the `::before` elements as `counter-set` and `counter-increment` together on the same element do not work (i.e., the counter is not incremented).

Due to the similarity of the code, I didn't perform much testing. But it seemed to work both in view and edit mode and with an explicitly set number.

Reply:
Btw., I did test on epiphany (Gnome Web) and it indeed doesn't seem to support `counter-set` and thus uses the old code.

Reply:
> Regarding number skipping, isn't this covered by :not([data-xwiki-rendering-protected]) in most of the selectors below?

Good, I mainly wanted to make sure this case had been considered.



> Btw., I did test on epiphany (Gnome Web) and it indeed doesn't seem to support `counter-set` and thus uses the old code.

Nice, that's a good sign this is a working solution. +1 to merge.